activerecord/test/cases/associations/has_one_associations_test.rb

```diff
@@ -37,6 +37,15 @@ def test_has_one
     end
   end
 
+  def test_has_one_with_exception
+    firm = companies(:first_firm)
+    firm.account.destroy!
+
+    assert_raises ActiveRecord::RecordNotFound do
+      firm.account!
+    end
```

Comment:
I can optimize this test by using a fixture that does not have an association already.

Reply:
Actually, on second thought, I think this is work keeping as-is. This ensures the new method reloads the association first. If we do not reload, we risk returning state data.